examples/c/ex_col_store.c

```diff
@@ -0,0 +1,536 @@
+        case 3:
+            strcpy(w.country, "NZD");
+            break;
+        case 4:
+            strcpy(w.country, "IND");
+            break;
+        case 5:
+            strcpy(w.country, "CHI");
+            break;
+        case 6:
+            strcpy(w.country, "RUS");
+            break;
+        default:
+            assert(false);
+        }
+
+        w_array[i] = w;
+    }
+}
+
+/*
+ * The function returns 0 when a valid min/max temperature can be calculated given the time range.
+ * If no records are found it will return WT_NOTFOUND, otherwise the program will crash if an
+ * internal error is encountered.
+ */
+static int
+find_min_and_max_temp(
+  WT_SESSION *session, uint16_t start_time, uint16_t end_time, int *min_temp, int *max_temp)
+{
+    WT_CURSOR *end_time_cursor, *join_cursor, *start_time_cursor;
+    uint64_t recno;
+    int exact;
+    int ret;
+    uint16_t hour;
+    uint8_t temp;
+
+    /* Open cursors needed by the join. */
+    error_check(
+      session->open_cursor(session, "join:table:weather(hour,temp)", NULL, NULL, &join_cursor));
+    error_check(
+      session->open_cursor(session, "index:weather:hour", NULL, NULL, &start_time_cursor));
+    error_check(session->open_cursor(session, "index:weather:hour", NULL, NULL, &end_time_cursor));
+
+    /*
+     * Select values WHERE (hour >= start AND hour <= end). Find the starting record closest to
+     * desired start time.
+     */
+    start_time_cursor->set_key(start_time_cursor, start_time);
+    error_check(start_time_cursor->search_near(start_time_cursor, &exact));
+    ret = 0;
+    if (exact == -1) {
+        ret = start_time_cursor->next(start_time_cursor);
+        if (ret == WT_NOTFOUND)
+            return ret;
+        else
+            error_check(ret);
+    }
+
+    error_check(session->join(session, join_cursor, start_time_cursor, "compare=ge"));
+
+    /* Find the ending record closest to desired end time. */
+    end_time_cursor->set_key(end_time_cursor, end_time);
+    error_check(end_time_cursor->search_near(end_time_cursor, &exact));
+    if (exact == 1) {
+        ret = end_time_cursor->prev(end_time_cursor);
+        if (ret == WT_NOTFOUND)
+            return ret;
+        else
+            error_check(ret);
+    }
+
+    error_check(session->join(session, join_cursor, end_time_cursor, "compare=le"));
+
+    /* Initialize minimum temperature and maximum temperature to temperature of the first record. */
+    ret = join_cursor->next(join_cursor);
+    if (ret == WT_NOTFOUND)
+        return ret;
+    else
+        error_check(ret);
+
+    error_check(join_cursor->get_key(join_cursor, &recno));
+    error_check(join_cursor->get_value(join_cursor, &hour, &temp));
+    *min_temp = temp;
+    *max_temp = temp;
+
+    /* Iterating through found records between start and end time to find the min & max temps. */
+    while ((ret = join_cursor->next(join_cursor)) == 0) {
+        error_check(join_cursor->get_key(join_cursor, &recno));
```

Comment:
I don't think there is a need to go a get_key, since we do not use a recno. You can do a get value without a get key.